sdk/include/sdk/IPubSubClient.h

```diff
@@ -126,6 +133,20 @@ class IPubSubClient {
      */
     virtual void publishOnTopic(const std::string& topic, const std::string& data) = 0;
 
+    /**
+     * @brief Publishes a message to the specified MQTT topic with a timeout in milliseconds for the
+     * publish to complete. Returns a status indicating whether the publish was successful, timed
+     * out, or failed.
+     *
+     * @param topic the MQTT topic to publish the message to
+     * @param data the payload to send as the message
+     * @param timeout_ms maximum time to wait for the publish to complete, in milliseconds
+     * @return PublishStatus indicating the result of the publish operation: Success, Timeout,
+     * Failure
+     */
+    virtual PublishStatus publishOnTopic(const std::string& topic, const std::string& data,
+                                         int timeout_ms) = 0;
```

Comment:
I'd prefer using some unsigned int to make clear that negative numbers are not useful.
```suggestion
                                         unsigned int timeout_ms) = 0;
```